Source/WebCore/page/EventHandler.cpp

```diff
@@ -1095,8 +1095,8 @@ void EventHandler::updateSelectionForMouseDrag(const HitTestResult& hitTestResul
         newSelection.expandUsingGranularity(m_frame->selection().granularity());
         if (!newSelection.isBaseFirst() && !oldSelection.isBaseFirst() && oldSelection.end() < newSelection.end())
             newSelection.setBase(oldSelection.end());
-        else if (newSelection.isBaseFirst() && !oldSelection.isBaseFirst() && oldSelection.start() < newSelection.start() && m_dragStartSelection) {
-            VisibleSelection dragStartSelection { *m_dragStartSelection };
+        else if (newSelection.isBaseFirst() && !oldSelection.isBaseFirst() && oldSelection.start() < newSelection.start() && m_dragStartSelection && m_dragStartSelection.value().start.container && m_dragStartSelection.value().end.container) {
+            VisibleSelection dragStartSelection { { BoundaryPoint(*(m_dragStartSelection.value().start.container), m_dragStartSelection.value().start.offset), BoundaryPoint(*(m_dragStartSelection.value().end.container), m_dragStartSelection.value().end.offset) } };
```

Comment:
This is hard to read. Could this be factored into a lambda or a private member function that computes all this stuff and has a descriptive name?

Reply:
If you were referring to just the line constructing the SimpleRange to pass into the VisibleSelection constructor, yes. I defined a private member in EventHandler called createSimpleRangeFromDragStartSelection().